c2rust-analyze/src/type_desc.rs

```diff
@@ -98,3 +99,169 @@ pub fn local_perms_to_desc<'tcx>(
         pointee_ty,
     }
 }
+
+pub fn perms_to_desc_with_pointee<'tcx>(
+    tcx: TyCtxt<'tcx>,
+    pointee_ty: Ty<'tcx>,
+    ptr_ty: Ty<'tcx>,
+    perms: PermissionSet,
+    flags: FlagSet,
+) -> TypeDesc<'tcx> {
+    let (own, qty) = if flags.contains(FlagSet::FIXED) {
+        unpack_pointer_type(tcx, ptr_ty, pointee_ty)
+    } else {
+        perms_to_own_and_qty(perms, flags)
+    };
+    TypeDesc {
+        own,
+        qty,
+        pointee_ty,
+    }
+}
+
+/// Unpack an existing `Ty` into its ownership, quantity, and pointee type.
```

Comment:
the docs don't seem to match up with the function which just unpacks into ownership and quantity. am i missing something?

Reply:
That's outdated - I'll fix it